File: challenges/credit-check/python/credit_check.py
def credit_check(string):
    credit_numbers = to_list(string)  # converts string to list of digits
    two_list = times_two(credit_numbers)  # doubles each digit in list
    final_list = sum_gt_9(two_list)  # numbers > 9 sum of its own digits
    sum = sum_all(final_list)  # get sum  of numbers in list

    if sum % 10 == 0:  # check validity
        return 'The number is valid!'

    return 'The number is invalid!'


def to_list(credit_card_numbers):
    digitList = [int(digit) for digit in credit_card_numbers]
    return digitList


def times_two(digit_list):
    credit_numbers = list()
    for i, num in enumerate(reversed(digit_list)):
        if i % 2 == 0:
            credit_numbers.insert(0, num)
        else:
            credit_numbers.insert(0, num * 2)
    return credit_numbers


def split_digits_sum(number):
    digitList = [int(digit) for digit in str(number)]
    sum = 0
    for digit in digitList:
        sum = sum + digit
    return sum


def sum_gt_9(array):
    final_list = list()
    for num in array:
        if num > 9:
            final_list.append(split_digits_sum(num))
        else:
            final_list.append(num)
    return final_list


def sum_all(array):
    sum = 0
    for num in array:
        sum = sum + num
    return sum

File: challenges/credit-check/js/creditCheck.js
exports.creditCheck = (string) => {
    const creditNumbers = exports.toList(string) // converts string to list of digits
    const toList = exports.timesTwo(creditNumbers) // doubles each digit in list
    const finalList = exports.sumGT9(toList) // numbers > 9 sum of its own digits
    const sum = exports.sumNumbers(finalList) // get sum  of numbers in list

    if (sum % 10 === 0) {
        return 'The number is valid!'
    }

    return 'The number is invalid!'
}

exports.toList = (creditCardNumbers) => {
    let digitList = []
    for (let i = 0; i < creditCardNumbers.length; i++) {
        let digit = parseInt(creditCardNumbers[i])
        digitList.push(digit)
    }
    return digitList
}

exports.timesTwo = (digitList) => {
    let creditNumbers = []
    for (let i = digitList.length - 1; i >= 0; i--) {
        if (i % 2 === 0) {
            creditNumbers.unshift(digitList[i] * 2)
        } else {
            creditNumbers.unshift(digitList[i])
        }
    }
    return creditNumbers
}

exports.splitDigitsSum = (number) => {
    let digitList = Array.from(number.toString()).map(Number)
    let sum = 0
    for (let i = 0; i < digitList.length; i++) {
        sum += digitList[i]
    }
    return sum
}

exports.sumGT9 = (array) => {
    let finalList = []
    for (let i = 0; i < array.length; i++) {
        if (array[i] > 9) {
            finalList.push(exports.splitDigitsSum(array[i]))
        } else {
            finalList.push(array[i])
        }
    }
    return finalList
}

exports.sumNumbers = (array) => {
    sum = 0
    for (let i = 0; i < array.length; i++) {
        sum += array[i]
    }
    return sum
}